src/adapter/src/catalog/state.rs

```diff
@@ -766,72 +765,38 @@ impl CatalogState {
         Some(url)
     }
 
-    /// Parse a SQL string into a catalog view item with only a limited
-    /// context.
-    #[mz_ore::instrument]
-    pub fn parse_view_item(&self, create_sql: String) -> Result<CatalogItem, anyhow::Error> {
-        let mut session_catalog = self.for_system_session();
-
-        // Enable catalog features that might be required during planning in
-        // [Catalog::open]. Existing catalog items might have been created while
-        // a specific feature flag was turned on, so we need to ensure that this
-        // is also the case during catalog rehydration in order to avoid panics.
-        //
-        // WARNING / CONTRACT:
-        // 1. Features used in this method that related to parsing / planning
-        //    should be `enable_for_item_parsing` set to `true`.
-        // 2. After this step, feature flag configuration must not be
-        //    overridden.
-        session_catalog.system_vars_mut().enable_for_item_parsing();
-
-        let stmt = mz_sql::parse::parse(&create_sql)?.into_element().ast;
-        let (stmt, resolved_ids) = mz_sql::names::resolve(&session_catalog, stmt)?;
-        let plan = mz_sql::plan::plan(
-            None,
-            &session_catalog,
-            stmt,
-            &Params::empty(),
-            &resolved_ids,
-        )?;
-        Ok(match plan {
-            Plan::CreateView(CreateViewPlan { view, .. }) => {
-                // Collect optimizer parameters.
-                let optimizer_config =
-                    optimize::OptimizerConfig::from(session_catalog.system_vars());
-
-                // Build an optimizer for this VIEW.
-                let mut optimizer = optimize::view::Optimizer::new(optimizer_config, None);
-
-                // HIR ⇒ MIR lowering and MIR ⇒ MIR optimization (local)
-                let raw_expr = view.expr;
-                let optimized_expr = optimizer.optimize(raw_expr.clone())?;
-
-                CatalogItem::View(View {
-                    create_sql: view.create_sql,
-                    raw_expr,
-                    desc: RelationDesc::new(optimized_expr.typ(), view.column_names),
-                    optimized_expr,
-                    conn_id: None,
-                    resolved_ids,
-                })
-            }
-            _ => bail!("Expected valid CREATE VIEW statement"),
-        })
-    }
```

Comment:
This was only ever used in a single test. So I updated the test to use an equivalent function and deleted this one.